crates/kernel/src/tasks/mod.rs

```diff
@@ -29,6 +30,12 @@ pub mod vm_host;
 
 pub type TaskId = usize;
 
+pub struct TaskHandle(pub TaskId, pub oneshot::Receiver<TaskWaiterResult>);
```

Comment:
Nit: This seems weird. If the fields are `pub`, then why the `fn task_id()`? More importantly: why a tuple-style `struct` instead of named fields?